src/librustc_mir/hair/cx/mod.rs

```diff
@@ -151,22 +151,13 @@ impl<'a, 'gcx, 'tcx> Cx<'a, 'gcx, 'tcx> {
 
     pub fn type_bit_size(
         &self,
-        ty: Ty<'tcx>,
+        ty: Ty,
     ) -> u64 {
-        let tcx = self.tcx.global_tcx();
-        let (ty, param_env) = self
-            .tcx
-            .lift_to_global(&(ty, self.param_env))
-            .unwrap_or_else(|| {
-            bug!("MIR: Cx::const_eval_literal({:?}, {:?}) got \
-                type with inference types/regions",
-                ty, self.param_env);
-        });
-        tcx
-            .layout_of(param_env.and(ty))
-            .expect("int layout")
-            .size
-            .bits()
+        match ty.sty {
+            TyInt(ity) => ity.bit_width(),
+            TyUint(uti) => uty.bit_width(),
```

Comment:
I don't expect this to work without a `tcx`... `usize` is a thing.

Reply:
these result in `none` if it's `usize`/`isize`, that's what the `unwrap_or_else` handles